lib/base-compiler.js

```diff
@@ -477,10 +477,10 @@ class BaseCompiler {
         /* if not defined, consider it true */
 
         if (gccDumpOptions.treeDump !== false) {
-            addOpts.push('-fdump-tree-all');
+            addOpts.push('-fdump-tree-all' + gccDumpOptions.dumpFlags);
```

Comment:
given the other conversation below, surely we need to remove the `-all` bit from this, yes?

Reply:
I'm just going to bow out on this - tried it locally and it appears to be working.  The GCC docs are _incredibly_ confusing on this subject...

Reply:
> given the other conversation below, surely we need to remove the `-all` bit from this, yes?

The answer would be no. The format is `-fdump-what-pass-options`. So `-fdump-tree-all-all` would produce dumps for all passes with all options enabled.